_posts/2018-04-09-synching-sia.md

```diff
@@ -0,0 +1,131 @@
+
+As mentioned in my previous article [Benchmarking Sia’s Blockchain Sync](https://blog.spaceduck.io/sia-blockchain-sync/) Sia spends an average of 55 minutes or 19.1% of its time notifying subscribers.  I submitted the pull request [#2809](https://github.com/NebulousLabs/Sia/pull/2809) so Sia only notifies subscribers if there are any.  By running Sia with only the `consensus` and `gateway` modules loaded there will be zero subscribers.
+
+My `siad` command with minimal modules loaded.
+
+```shell
+./siad -M cg -d /root/Sia/
+```
+
+With zero subscribers Sia skips the notify subscribers functions and shaves 55 minutes off the synchronization time.  Note that with only the `consensus` and `gateway` modules loaded you cannot rent, host, or use the Sia wallet.
+
+## Reduce RPC Timeouts
+
+Sia has long timeouts when waiting for blocks to download.  I’ve submitted the pull request [#2922](https://github.com/NebulousLabs/Sia/pull/2922) to reduce some of the long timeouts.  In order to sync in 73 minutes, I dropped the timeouts in [syncronize.go](https://github.com/NebulousLabs/Sia/blob/master/modules/consensus/synchronize.go) **WAY** down.  I first tried setting `relayHeaderTimeout`, `sendBlkTimeout` and `sendBlocksTimeout` to 2 seconds, this was too low, I was timing out on all my peers and couldn’t download any blocks.  After bumping `sendBlkTimeout` and `sendBlocksTimeout` to 10 seconds I was able to stay connected to most peers and download the blocks very quickly.  Any peers that could not deliver my batch of 10 blocks in 10 seconds or less was disconnected and Sia would look for new peers.
+
+Note that 10-second timeouts are not very friendly to the Sia network and 10 seconds is not feasible for users with slower internet connections.  For example, the average Tor user or 3rd world Internet user would not be able to sync Sia with `sendBlocksTimeout` set to 10 seconds.  For this reason, the default `sendBlocksTimeout` in my pull request has been set at 180 seconds to allow for users with slower internet connections.  The Sia devs and I think this is a good balance of accommodating users with slower connections, while not wasting too much time waiting for unresponsive peers.
+
+My 73-minute sync only spent 168 seconds downloading blocks from peers.  This number is actually slightly confusing to me since if Sia downloaded 10GB of blocks it exceeded my advertised home internet speed.  A couple of things could have happened here:
+The downloaded blocks could be less than 10GB in size when sent over the Internet, I never measured how much data I actually received from my NIC.
+I could have a mistake in my timing function causing the elapsed download time to be reported wrong.
+My ISP could allow me to burst into higher speed tiers during low network congestion.
+
+Regardless of why my download speed seems off, I was able to sync from block 0 to current height in 73 minutes.
+
+The timeouts I set in `synchronize.go` for my 73-minute sync.
```

Comment:
Can we link to a diff of your test code against the latest commit? 

Each of the other optimizations includes instructions for how the user can replicate the results, so I think these should too. Something kind of like:

```shell
go get -u github.com/NebulousLabs/Sia/...
cd $GOPATH/src/github.com/NebulousLabs/Sia
git remote add tbenz9 https://github.com/tbenz9/Sia.git
# Not sure the exact syntax but check out the speedup code from your branch
git checkout -b reduce-timeouts tbenz9/reduce-timeouts
make release-std
$GOPATH/bin/siad ...
```